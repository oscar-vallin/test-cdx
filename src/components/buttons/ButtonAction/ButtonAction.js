import PropTypes from 'prop-types';
import { StyledButtonAction } from './ButtonAction.styles';

const buttonIcons = {
  edit: 'Edit',
  sort: 'Sort',
  eye: 'RedEye',
  chromeBack: 'ChromeBack',
  chromeNext: 'ChromeBackMirrored',
  up: 'Up',
  down: 'Down',
  today: 'GotoToday',
  asc: 'SortUp',
  desc: 'SortDown',
};

const ButtonAction = ({ id = '__Button', children, icon, disabled = false, onClick, ...props }) => {
  const _icon = { iconName: buttonIcons[icon] };

  return (
    <StyledButtonAction id={id} disabled={disabled} onClick={onClick} iconProps={_icon} {...props}>
      {children}
    </StyledButtonAction>
  );
};

ButtonAction.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node,
  primary: PropTypes.bool,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  icon: PropTypes.string,
};

export { ButtonAction };
