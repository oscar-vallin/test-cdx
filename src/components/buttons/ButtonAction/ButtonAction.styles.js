import styled from 'styled-components';
import { ActionButton } from '@fluentui/react';

export const StyledButtonAction = styled(ActionButton)`
  color: ${({ theme }) => theme.colors.themePrimary};
`;
