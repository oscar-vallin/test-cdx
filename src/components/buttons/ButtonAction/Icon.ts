import { IIconProps } from "office-ui-fabric-react";

export const getIcon = (name: string) => {
  const addFriendIcon: IIconProps = { iconName: "AddFriend" };

  return addFriendIcon;
};
