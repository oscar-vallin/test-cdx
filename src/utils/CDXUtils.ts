import { format } from 'date-fns';

function isDateTimeValid(d?: Date | null): boolean {
  // Check if this date has a valid time value
  /* eslint-disable no-self-compare */
  return !!(d && d.getTime() === d.getTime());
}

function yyyyMMdd(date?: Date | null): string {
  if (date) {
    return format(date, 'yyyy-MM-dd');
  }
  return '';
}

function yyyyMMdda(date?: Date | null): string {
  if (date) {
    return format(date, 'yyyy-MM-dd a');
  }
  return '';
}

function getEnumByValue(enumType, enumValue?: string) {
  const key = Object.keys(enumType).find((x) => enumType[x] === enumValue);
  if (key) {
    return enumType[key];
  }
  return null;
}

export { isDateTimeValid, yyyyMMdd, getEnumByValue, yyyyMMdda };
