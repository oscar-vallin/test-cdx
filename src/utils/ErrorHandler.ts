import { useHistory } from 'react-router-dom';
import { ROUTES } from 'src/data/constants/RouteConstants';
import { useSessionStore } from 'src/store/SessionStore';
import { useNotification } from 'src/hooks/useNotification';
import { useApplicationStore } from 'src/store/ApplicationStore';
import { useLogoutUseCase } from 'src/use-cases/Authentication';

export const ErrorHandler = () => {
  const Toast = useNotification();
  const SessionStore = useSessionStore();
  const ApplicationStore = useApplicationStore();
  const { performUserLogout } = useLogoutUseCase();
  const history = useHistory();

  return (error?: any) => {
    if (error) {
      if (error?.networkError?.statusCode === 403) {
        SessionStore.setGlobalError('Your session has expired please login again.');
        performUserLogout();
      } else {
        const { message } = error;
        const { extensions = null } = error?.graphQLErrors?.shift() || {};

        if (extensions) {
          if (extensions.errorSubType === 'NEED_AUTH') {
            SessionStore.setGlobalError(message);
            performUserLogout();
          } else if (extensions.errorSubType === 'INSUFFICIENT_PRIVILEGES') {
            history.push(ROUTES.ROUTE_UNAUTHORIZED.URL);
          } else {
            Toast.error({ text: 'An internal server error has occurred. Please contact your administrator.' });
          }
        } else if (message === 'Failed to fetch') {
          ApplicationStore.setIsOffline(true);
        } else {
          Toast.error({ text: 'An internal server error has occurred. Please contact your administrator.' });
        }
      }
    }
  };
};
