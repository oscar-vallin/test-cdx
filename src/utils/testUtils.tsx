import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '../styles/themes/index';
import { theme } from '../styles/themes/theme';

// eslint-disable-next-line react/prop-types
const LightTheme = { ...theme, ...defaultTheme };
export const mountWithTheme = (children: any) => mount(<ThemeProvider theme={LightTheme}>{children}</ThemeProvider>);

export const shallowWithTheme = (children: any) =>
  shallow(<ThemeProvider theme={LightTheme}>{children}</ThemeProvider>);

export const renderWithTheme = (children: any) => render(<ThemeProvider theme={LightTheme}>{children}</ThemeProvider>);

export * from '@testing-library/react';

export const randomString = (length: number) => {
  let result = '';
  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
  const charactersLength = characters.length;
  for (let i = 0; i < length; i++) {
    result += characters.charAt(Math.floor(Math.random() * charactersLength));
  }
  return result;
};
