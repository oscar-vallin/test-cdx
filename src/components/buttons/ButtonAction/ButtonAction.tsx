import { ReactNode } from 'react';
import { StyledButtonAction } from './ButtonAction.styles';

const defaultProps = {
  id: '',
  icon: '',
  disabled: false,
  onClick: () => null,
  iconProps: '',
};

type ButtonActionProps = {
  id?: string;
  children: ReactNode;
  disabled: boolean;
  onClick?: () => null;
  icon: string;
  iconProps?: string;
} & typeof defaultProps;

const buttonIcons = {
  edit: 'Edit',
  sort: 'Sort',
  eye: 'RedEye',
  chromeBack: 'ChromeBack',
  chromeNext: 'ChromeBackMirrored',
  up: 'Up',
  down: 'Down',
  today: 'GotoToday',
  asc: 'SortUp',
  desc: 'SortDown',
};

const ButtonAction = ({ id, children, icon, disabled = false, onClick, ...props }: ButtonActionProps) => {
  const _icon = { iconName: buttonIcons[icon] };

  return (
    <StyledButtonAction id={id} disabled={disabled} onClick={onClick} iconProps={_icon} {...props}>
      {children}
    </StyledButtonAction>
  );
};

// ButtonAction.propTypes = {
//   id: PropTypes.string,
//   children: PropTypes.node,
//   primary: PropTypes.bool,
//   disabled: PropTypes.bool,
//   onClick: PropTypes.func,
//   icon: PropTypes.string,
// };

export { ButtonAction };
