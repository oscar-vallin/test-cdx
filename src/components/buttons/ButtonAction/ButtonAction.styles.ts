import styled from 'styled-components';
import { ActionButton } from '@fluentui/react';
import { IIconProps } from 'office-ui-fabric-react/lib-commonjs/Icon';
import { IconType } from 'recharts/types/component/DefaultLegendContent';
import { IImageProps } from 'office-ui-fabric-react/lib-commonjs/Image';

export const StyledButtonAction = styled(ActionButton)`
  color: ${({ theme }) => theme.colors.themePrimary};
`;
