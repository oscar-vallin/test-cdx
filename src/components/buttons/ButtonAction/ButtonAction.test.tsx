import { shallowWithTheme } from '../../../../src/utils/testUtils';
import { ButtonAction } from './ButtonAction';

const defaultProps = {
  id: '',
  icon: '',
  disabled: false,
  onClick: () => null,
  iconProps: '',
};

describe('ButtonAction', () => {
  const mockFn = jest.fn();
  const tree = shallowWithTheme(
    <ButtonAction {...defaultProps} onClick={mockFn} iconProps="edit">
      Testing ButtonAction
    </ButtonAction>
  );

  it('Should be defined', () => {
    expect(ButtonAction).toBeDefined();
  });

  it('Should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Should have a click function', () => {
    tree.children().simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  it('Should have an icon', () => {
    expect(tree.children().props().iconProps).toEqual('edit');
  });

  it('Should renders children', () => {
    expect(tree.children().contains('Testing ButtonAction')).toEqual(true);
  });

  it('Should renders children when passed in', () => {
    const wrapper = shallowWithTheme(
      <ButtonAction {...defaultProps}>
        <div className="children" />
      </ButtonAction>
    );
    expect(wrapper.contains(<div className="children" />)).toEqual(true);
  });

  it('Test styled ButtonAction component', () => {
    const wrapper = shallowWithTheme(
      <ButtonAction {...defaultProps}>
        <div className="children" />
      </ButtonAction>
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
