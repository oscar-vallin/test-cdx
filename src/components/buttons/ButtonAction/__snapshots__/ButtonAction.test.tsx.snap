// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAction Should render correctly 1`] = `ShallowWrapper {}`;
